apis/python/tests/test_dataframe.py

```diff
@@ -139,11 +139,17 @@ def test_dataframe_with_enumeration(tmp_path):
     ) as sdf:
         data = {}
         data["soma_joinid"] = [0, 1, 2, 3, 4]
-        data["foo"] = [2, 1, 2, 1, 0]
-        data["bar"] = [0, 1, 1, 0, 1]
-        sdf.write(pa.Table.from_pydict(data))
-        assert sdf.enumeration("foo") == enums["enmr1"]
-        assert sdf.enumeration("bar") == enums["enmr2"]
+        if downgrade:
+            data["foo"] = ["a", "bb", "ccc", "bb", "a"]
+            data["bar"] = ["cat", "dog", "cat", "cat", "cat"]
+            with pytest.raises(ValueError):
+                sdf.write(pa.Table.from_pydict(data))
+        else:
+            data["foo"] = pd.Categorical(["a", "bb", "ccc", "bb", "a"])
+            data["bar"] = pd.Categorical(["cat", "dog", "cat", "cat", "cat"])
+            sdf.write(pa.Table.from_pydict(data))
+            assert sdf.enumeration("foo") == enums["enmr1"]
+            assert sdf.enumeration("bar") == enums["enmr2"]
```

Comment:
I don't think this needs to be parameterized.

```suggestion
          data["foo"] = ["a", "bb", "ccc", "bb", "a"]
          data["bar"] = ["cat", "dog", "cat", "cat", "cat"]
          with pytest.raises(ValueError):
              sdf.write(pa.Table.from_pydict(data))

          data["foo"] = pd.Categorical(["a", "bb", "ccc", "bb", "a"])
          data["bar"] = pd.Categorical(["cat", "dog", "cat", "cat", "cat"])
          sdf.write(pa.Table.from_pydict(data))
          assert sdf.enumeration("foo") == enums["enmr1"]
          assert sdf.enumeration("bar") == enums["enmr2"]
```